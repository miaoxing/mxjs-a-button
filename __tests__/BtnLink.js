import {BtnLink} from '../';
import {MemoryRouter} from 'react-router';
import {render} from '@testing-library/react';

describe('BtnLink', () => {
  test('renders correctly', () => {
    const {container} = render(<MemoryRouter>
      <BtnLink href="foo">bar</BtnLink>
    </MemoryRouter>);

    expect(container).toMatchSnapshot();
  });

  test('text', () => {
    const {getByTestId} = render(<MemoryRouter>
      <BtnLink data-testid="foo" href="foo">bar</BtnLink>
    </MemoryRouter>);

    expect(getByTestId('foo').innerHTML).toBe('bar');
  });
});
