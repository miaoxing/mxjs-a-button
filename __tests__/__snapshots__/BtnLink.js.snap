// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnLink renders correctly 1`] = `
<div>
  <a
    class="ant-btn"
    href="/"
  >
    <span>
      bar
    </span>
  </a>
</div>
`;

exports[`BtnLink text 1`] = `
<a
  class="ant-btn"
  data-testid="foo"
  href="/"
>
  <span>
    bar
  </span>
</a>
`;
